db/data/20210712154115_remove_orphaned_historical_events.rb

```diff
@@ -0,0 +1,4 @@
+# Run me with `rails runner db/data/20210712154115_remove_orphaned_historical_events.rb`
+#
+orphaned_events = HistoricalEvent.all.select { |event| Activity.find_by(id: event.activity_id).nil? }
```

Comment:
It's not important here, in a one-off script, but I think you can do what you want much more efficient in SQL with an "inner join":

```
HistoricalEvent.joins(:activity).count
   (29.9ms)  SELECT COUNT(*) FROM "historical_events" INNER JOIN "activities" ON "activities"."id" = "historical_events"."activity_id"
=> 2143
```
as opposed to a left outer joins which would return all historical events including those whose activity no longer exists?

```
 HistoricalEvent.left_joins(:activity).count
   (1.4ms)  SELECT COUNT(*) FROM "historical_events" LEFT OUTER JOIN "activities" ON "activities"."id" = "historical_events"."activity_id"
=> 2153
```

i.e. there are 10 historical records orphaned?

See https://blog.codinghorror.com/a-visual-explanation-of-sql-joins/ for a great visualisation of these joins....

Reply:
🙏 Thanks @edavey I really appreciate deeper learning like this!